### DISCUSSION TITLE
Playback rate of storyboard samples doesn't change dynamically during gameplay

### DISCUSSION THREAD

**Discussion by hlysine** [ppy/osu#17053] [Q&A] [answered]
This bug affects mods having variable rates (currently Wind Up and Wind Down). 

https://github.com/ppy/osu/blob/c197e27e33bc77da7b96990dc10d2677eda38cc4/osu.Game/Rulesets/Mods/ModTimeRamp.cs#L64-L67

Although the adjustment is applied with the `SpeedChange` bindable, the sample rate always stays at the initial value of the bindable and doesn't change during gameplay. This may be due to `DrawableStoryboardSample` getting different mod instances than other parts of gameplay.

This bug is particularly noticeable in https://osu.ppy.sh/beatmapsets/602861#mania/1273379 with Wind Up / Wind Down and "Adjust pitch" turned on. Some parts of the map are completely messed up due to incorrect pitch.

**Comment by frenzibyte** [comment, marked as answer]
#17055
